--- upstream/runCommand (vendor/k8s.io/kubernetes/cmd/kube-scheduler/app/server.go)
+++ local/runCommand (pkg-kni/app/sched_command.go)
@@ -1,4 +1,4 @@
-func runCommand(cmd *cobra.Command, opts *options.Options, registryOptions ...Option) error {
+func runCommand(cmd *cobra.Command, opts *options.Options, registryOptions ...app.Option) error {
 	verflag.PrintAndExitIfRequested()
 	fg := opts.ComponentGlobalsRegistry.FeatureGateFor(basecompatibility.DefaultKubeComponent)
 	// Activate logging as soon as possible, after that
@@ -17,7 +17,7 @@
 		cancel()
 	}()
 
-	cc, sched, err := Setup(ctx, opts, registryOptions...)
+	cc, sched, err := app.Setup(ctx, opts, registryOptions...)
 	if err != nil {
 		return err
 	}
@@ -26,5 +26,5 @@
 	// add component version metrics
 	opts.ComponentGlobalsRegistry.AddMetrics()
 
-	return Run(ctx, cc, sched)
+	return run(ctx, cc, sched, customServe)
 }
